src/core/QuoridorGame.cs

```diff
@@ -50,7 +94,103 @@ public void UndoMove(Move move)
         {
-            return 0;
+            throw new NotImplementedException();
+        }
+
+        public void GetWhiteCoordinates(ref int row, ref int column)
+        {
+            row = white.Row;
+            column = white.Column;
+        }
+
+        public void GetBlackCoordinates(ref int row, ref int column)
+        {
+            row = black.Row;
+            column= black.Column;
+        }
+
+        public int Dimension { get; }
+
+        /// <summary>
+        /// Executes a given player movement in this state. If the move has invalid parameters, throws an
+        /// InvalidMoveException.
+        /// </summary>
+        /// <param name="move">The move desribing the player movement</param>
+        private void movePlayer(QuoridorMove move)
+        {
+            // Checking if values are inside bounds
+            if (move.Column < 0 || move.Column >= dimension) throw new InvalidMoveException("Coordinates out of bounds");
+            if (move.Row < 0 || move.Row >= dimension) throw new InvalidMoveException("Coordinates out of bounds");
+
+            // Check if any walls make the move impossible
+            QuoridorPlayer targetPlayer = getTargetPlayer(move.IsWhitePlayer);
+            Debug.Assert(targetPlayer is not null);
+
+            int deltaRow = move.Row - targetPlayer.Row;
+            int deltaColumn = move.Column - targetPlayer.Column;
+
+            if (deltaRow == 0 && board.CheckWallHorizontal(targetPlayer.Row, targetPlayer.Column, move.Column)) 
+                throw new InvalidMoveException("Wall is blocking player move");
+            if (deltaColumn == 0 && board.CheckWallVertical(targetPlayer.Column, targetPlayer.Row, move.Row))
+                throw new InvalidMoveException("Wall is blocking player move");
+
+            // Checking if another player is already located on destination coordinates
+            if (white.Row == move.Row && white.Column == move.Column) throw new InvalidMoveException("This position is occupied");
+            if (black.Row == move.Row && black.Column == move.Column) throw new InvalidMoveException("This position is occupied");
+
+            // Finally execute the move
+            targetPlayer.Row = move.Row;
+            targetPlayer.Column = move.Column;
+
+            gameHistory.Add(move);
+        } 
+
+        /// <summary>
+        /// Places a wall according to the parameters desribed by the given move
+        /// If the wall placement is invalid throws an InvalidMoveException
+        /// </summary>
+        /// <param name="move">The move describing the wall placement</param>
+        private void placeWall(QuoridorMove move)
+        {
+            // Checking if values are inside bounds
+            if (move.Column < 0 || move.Column >= dimension - 1) throw new InvalidMoveException("Coordinates out of bounds");
+            if (move.Row < 1 || move.Row >= dimension) throw new InvalidMoveException("Coordinates out of bounds");
+
```

Comment:
Add a comment with the following line to keep that in mind:
_We also have to check if newly placed wall forms a cross with an already placed one with opposite orientation_